src/nbody/nbody_utils.hpp

```diff
@@ -34,10 +36,48 @@ namespace NBody {
 extern void reb_extra_forces(struct reb_simulation *rsim);
 extern int collision_resolution(struct reb_simulation *const r, struct reb_collision c);
 
+class RebSim {
+ public:
+  // Constructor to initialize the shared_ptr
+  RebSim() : reb_sim(reb_simulation_create(), reb_sim_deleter) {}
+
+  // Constructor that accepts a preexisting pointer
+  RebSim(const RebSim &existing_reb_sim)
+      : reb_sim(reb_simulation_copy(existing_reb_sim), reb_sim_deleter) {}
+
+  // Constructor that accepts a rebound filename
+  RebSim(std::string reb_filename)
+      : reb_sim(reb_simulation_create_from_file(reb_filename.data(), -1),
+                reb_sim_deleter) {}
+
+  // Function to get a reference to the shared_ptr
+  struct reb_simulation *get() const {
+    PARTHENON_REQUIRE(reb_sim, "Internal pointer is null!");
+    return reb_sim.get();
+  }
+
+  // Conversion operator to return the raw pointer
+  operator struct reb_simulation *() const { return reb_sim.get(); }
+
+  // Overloaded -> operator
+  reb_simulation *operator->() const { return reb_sim.get(); }
```

Comment:
Some C++ trickery that lets us to `RebSim->integrator =` and `reb_simulation_function(RebSim, ...)` without the extra `.get()`s